### ISSUE TITLE
Redis backup failing in 3.2.1 - works in 3.1.3

### ISSUE THREAD

**Issue by mrplow** [tiredofit/docker-db-backup#119]
### Summary

Backing up my redis database is failing on the latest version, it works using tag 3.1.3

### Steps to reproduce

backup a redis database using 3.2.1?

### Relevant logs and/or screenshots
<details>
<summary>Working log from 3.1.3:</summary>

```$ docker-compose exec db-backup-redis backup-now
** Performing Manual Backup
+ source /assets/functions/10-db-backup
+ source /assets/defaults/10-db-backup
++ BACKUP_LOCATION=FILESYSTEM
++ CHECKSUM=MD5
++ COMPRESSION=XZ
++ COMPRESSION_LEVEL=9
++ DB_DUMP_BEGIN=0345
++ DB_DUMP_FREQ=1440
++ DB_DUMP_TARGET=/backup
++ ENABLE_CHECKSUM=TRUE
++ ENABLE_PARALLEL_COMPRESSION=TRUE
++ MANUAL_RUN_FOREVER=TRUE
++ MODE=AUTO
++ MYSQL_MAX_ALLOWED_PACKET=512M
++ MYSQL_SINGLE_TRANSACTION=TRUE
++ MYSQL_STORED_PROCEDURES=TRUE
+++ nproc
++ PARALLEL_COMPRESSION_THREADS=4
++ S3_CERT_SKIP_VERIFY=TRUE
++ S3_PROTOCOL=https
++ SIZE_VALUE=bytes
++ SPLIT_DB=TRUE
++ TEMP_LOCATION=/tmp/backups
++ dbhost=redis
++ dbname=
++ dbpass=
++ dbtype=redis
++ dbuser=
+ PROCESS_NAME=db-backup
+ CONTAINER_LOG_LEVEL=DEBUG
+ bootstrap_variables
+ case "${dbtype,,}" in
+ dbtype=redis
+ dbport=6379
+ [[ -n '' ]]
+ [[ -n '' ]]
+ '[' filesystem = s3 ']'
+ '[' filesystem = minio ']'
+ case "$dbtype" in
+ [[ -n '' ]]
+ '[' auto = manual ']'
+ '[' manual = manual ']'
+ DB_DUMP_BEGIN=+0
+ manual=TRUE
+ print_debug 'Detected Manual Mode'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:46 [45m[DEBUG][49m /etc/services.available/10-db-backup/run ** [db-backup] Detected Manual Mode
+ true
+ mkdir -p /tmp/backups
++ date +%s
+ backup_start_time=1649962846
++ date '+%Y-%m-%d %T %Z'
+ print_debug 'Backup routines started time: 2022-04-14 12:00:46 PDT'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:46 [45m[DEBUG][49m /etc/services.available/10-db-backup/run ** [db-backup] Backup routines started time: 2022-04-14 12:00:46 PDT
+ case "${dbtype,,}" in
+ check_availability
+ case "$dbtype" in
+ COUNTER=0
+ nc -z redis 6379
+ backup_redis
+ pre_dbbackup
++ date +%s
+ dbbackup_start_time=1649962846
++ date +%Y%m%d-%H%M%S
+ now=20220414-120046
++ date +%H:%M:%S
+ now_time=12:00:46
++ date +%Y-%m-%d
+ now_date=2022-04-14
+ target=redis__redis_20220414-120046.sql
+ print_notice 'Dumping Redis - Flushing Redis Cache First'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:46 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Dumping Redis - Flushing Redis Cache First
+ target=redis_all_redis_20220414-120046.rdb
+ echo bgsave
+ redis-cli -h redis -p 6379 --rdb /tmp/backups/redis_all_redis_20220414-120046.rdb
sending REPLCONF capa eof
sending REPLCONF rdb-only 1
SYNC sent to master, writing 534098 bytes to '/tmp/backups/redis_all_redis_20220414-120046.rdb'
Transfer finished with success.
+ sleep 10
+ try=5
+ '[' 5 -gt 0 ']'
++ echo 'info Persistence'
++ redis-cli -h redis -p 6379
++ awk '/rdb_bgsave_in_progress:0/{print "saved"}'
+ saved=saved
++ echo 'info Persistence'
++ redis-cli -h redis -p 6379
++ awk '/rdb_last_bgsave_status:ok/{print "ok"}'
+ ok=ok
+ [[ saved = \s\a\v\e\d ]]
+ [[ ok = \o\k ]]
+ print_notice 'Redis Backup Complete'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:56 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Redis Backup Complete
+ break
+ target_original=redis_all_redis_20220414-120046.rdb
+ compression
+ var_false TRUE
+ '[' true = false ']'
+ '[' true = no ']'
+ case "${COMPRESSION,,}" in
+ compress_cmd='pixz -9 -p 4 '
+ compression_type=xzip
+ target=redis_all_redis_20220414-120046.rdb.xz
+ case "${CONTAINER_LOG_LEVEL,,}" in
+ '[' xzip = none ']'
+ compression_string='and compressing with '\''xzip:9'\'' with '\''4'\'' threads'
+ pixz -9 -p 4 /tmp/backups/redis_all_redis_20220414-120046.rdb
+ generate_checksum
+ var_true TRUE
+ '[' true = true ']'
+ '[' '' = 0 ']'
+ print_warn 'Skipping Checksum creation because backup did not complete successfully'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:56 [44m[WARN][49m /etc/services.available/10-db-backup/run ** [db-backup] Skipping Checksum creation because backup did not complete successfully
+ move_dbbackup
+ case "$SIZE_VALUE" in
+ SIZE_VALUE=1
+ '[' 1 = 1 ']'
++ stat -c%s /tmp/backups/redis_all_redis_20220414-120046.rdb.xz
+ FILESIZE=31300
+ print_notice 'Backup of redis_all_redis_20220414-120046.rdb.xz created with the size of 31300 bytes'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:56 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Backup of redis_all_redis_20220414-120046.rdb.xz created with the size of 31300 bytes
+ case "${BACKUP_LOCATION,,}" in
+ print_debug 'Moving backup to filesystem'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:56 [45m[DEBUG][49m /etc/services.available/10-db-backup/run ** [db-backup] Moving backup to filesystem
+ mkdir -p /backup
+ mv '/tmp/backups/*.' /backup/
mv: can't rename '/tmp/backups/*.': No such file or directory
+ mv /tmp/backups/redis_all_redis_20220414-120046.rdb.xz /backup/redis_all_redis_20220414-120046.rdb.xz
+ post_dbbackup all
++ date +%s
+ dbbackup_finish_time=1649962856
++ echo 10
+ dbbackup_total_time=10
+ var_true TRUE
+ '[' true = true ']'
+ print_notice 'Sending Backup Statistics to Zabbix'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:56 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Sending Backup Statistics to Zabbix
++ stat -c%s /backup/redis_all_redis_20220414-120046.rdb.xz
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.size -o 31300
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.size -o 31300
Sending failed. Use option -vv for more detailed output.
++ date -r /backup/redis_all_redis_20220414-120046.rdb.xz +%s
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.datetime -o 1649962856
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.datetime -o 1649962856
Sending failed. Use option -vv for more detailed output.
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.status -o ''
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.status -o ''
Sending failed. Use option -vv for more detailed output.
++ echo 10
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.backup_duration -o 10
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.backup_duration -o 10
Sending failed. Use option -vv for more detailed output.
+ '[' -n '' ']'
+ '[' -d /assets/custom-scripts/ ']'
++ echo 10
++ awk '{printf "Hours: %d Minutes: %02d Seconds: %02d", $1/3600, ($1/60)%60, $1%60}'
+ print_notice 'DB Backup for '\''all'\'' time taken: Hours: 0 Minutes: 00 Seconds: 10'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:57 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] DB Backup for 'all' time taken: Hours: 0 Minutes: 00 Seconds: 10
++ date +%s
+ backup_finish_time=1649962857
++ echo 11
+ backup_total_time=11
+ '[' -z '' ']'
+ master_exit_code=0
++ date -d @1649962857 '+%Y-%m-%d %T %Z'
+ print_info 'Backup routines finish time: 2022-04-14 12:00:57 PDT with overall exit code 0'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:57 [42m[INFO][49m /etc/services.available/10-db-backup/run ** [db-backup] Backup routines finish time: 2022-04-14 12:00:57 PDT with overall exit code 0
++ echo 11
++ awk '{printf "Hours: %d Minutes: %02d Seconds: %02d", $1/3600, ($1/60)%60, $1%60}'
+ print_notice 'Backup routines time taken: Hours: 0 Minutes: 00 Seconds: 11'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:57 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Backup routines time taken: Hours: 0 Minutes: 00 Seconds: 11
+ cleanup_old_data
+ '[' -n 8640 ']'
+ '[' 0 '!=' 1 ']'
+ print_info 'Cleaning up old backups'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:57 [42m[INFO][49m /etc/services.available/10-db-backup/run ** [db-backup] Cleaning up old backups
+ mkdir -p /backup
+ find /backup/ -mmin +8640 -iname '*' -exec rm '{}' ';'
+ var_true TRUE
+ '[' true = true ']'
+ print_debug 'Exitting due to manual mode'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:00:57 [45m[DEBUG][49m /etc/services.available/10-db-backup/run ** [db-backup] Exitting due to manual mode
+ exit 0
```
</details>
<details>
<summary>Broken log from 3.2.1:</summary>

```$ docker-compose exec db-backup-redis backup-now
** Performing Manual Backup
+ source /assets/functions/10-db-backup
+ source /assets/defaults/10-db-backup
++ BACKUP_LOCATION=FILESYSTEM
++ CHECKSUM=MD5
++ COMPRESSION=XZ
++ COMPRESSION_LEVEL=9
++ DB_DUMP_BEGIN=0345
++ DB_DUMP_FREQ=1440
++ DB_DUMP_TARGET=/backup
++ ENABLE_CHECKSUM=TRUE
++ ENABLE_PARALLEL_COMPRESSION=TRUE
++ MANUAL_RUN_FOREVER=TRUE
++ MODE=AUTO
++ MYSQL_MAX_ALLOWED_PACKET=512M
++ MYSQL_SINGLE_TRANSACTION=TRUE
++ MYSQL_STORED_PROCEDURES=TRUE
+++ nproc
++ PARALLEL_COMPRESSION_THREADS=4
++ S3_CERT_SKIP_VERIFY=TRUE
++ S3_PROTOCOL=https
++ SIZE_VALUE=bytes
++ SPLIT_DB=TRUE
++ TEMP_LOCATION=/tmp/backups
++ dbhost=redis
++ dbname=
++ dbpass=
++ dbtype=redis
++ dbuser=
+ PROCESS_NAME=db-backup
+ CONTAINER_LOG_LEVEL=DEBUG
+ bootstrap_variables
+ case "${dbtype,,}" in
+ dbtype=redis
+ dbport=6379
+ [[ -n '' ]]
+ [[ -n '' ]]
+ '[' filesystem = s3 ']'
+ '[' filesystem = minio ']'
+ case "$dbtype" in
+ [[ -n '' ]]
+ '[' auto = manual ']'
+ '[' manual = manual ']'
+ DB_DUMP_BEGIN=+0
+ manual=TRUE
+ print_debug 'Detected Manual Mode'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:35 [45m[DEBUG][49m /etc/services.available/10-db-backup/run ** [db-backup] Detected Manual Mode
+ true
+ mkdir -p /tmp/backups
++ date +%s
+ backup_start_time=1649962895
++ date '+%Y-%m-%d %T %Z'
+ print_debug 'Backup routines started time: 2022-04-14 12:01:35 PDT'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:35 [45m[DEBUG][49m /etc/services.available/10-db-backup/run ** [db-backup] Backup routines started time: 2022-04-14 12:01:35 PDT
+ case "${dbtype,,}" in
+ check_availability
+ case "$dbtype" in
+ counter=0
+ nc -z redis 6379
+ backup_redis
+ pre_dbbackup
++ date +%s
+ dbbackup_start_time=1649962895
++ date +%Y%m%d-%H%M%S
+ now=20220414-120135
++ date +%H:%M:%S
+ now_time=12:01:35
++ date +%Y-%m-%d
+ now_date=2022-04-14
+ target=redis__redis_20220414-120135.sql
+ print_notice 'Dumping Redis - Flushing Redis Cache First'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:35 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Dumping Redis - Flushing Redis Cache First
+ target=redis_all_redis_20220414-120135.rdb
+ echo bgsave
+ redis-cli -h redis -p 6379 --rdb /tmp/backups/redis_all_redis_20220414-120135.rdb
sending REPLCONF capa eof
sending REPLCONF rdb-only 1
SYNC sent to master, writing 534098 bytes to '/tmp/backups/redis_all_redis_20220414-120135.rdb'
Transfer finished with success.
+ sleep 10
+ try=5
+ '[' 5 -gt 0 ']'
++ echo 'info Persistence'
++ redis-cli -h redis -p 6379
++ awk '/rdb_bgsave_in_progress:0/{print "saved"}'
+ saved=saved
++ echo 'info Persistence'
++ redis-cli -h redis -p 6379
++ awk '/rdb_last_bgsave_status:ok/{print "ok"}'
+ ok=ok
+ [[ saved = \s\a\v\e\d ]]
+ [[ ok = \o\k ]]
+ print_notice 'Redis Backup Complete'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:45 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Redis Backup Complete
+ break
+ target_original=redis_all_redis_20220414-120135.rdb
+ compression
+ var_false TRUE
+ '[' true = false ']'
+ '[' true = no ']'
+ case "${COMPRESSION,,}" in
+ compress_cmd='pixz -9 -p 4 '
+ compression_type=xzip
+ dir_compress_cmd='pixz -9 -p 4 '
+ extension=.xz
+ target_dir=redis_all_redis_20220414-120135.rdb
+ target=redis_all_redis_20220414-120135.rdb.xz
+ case "${CONTAINER_LOG_LEVEL,,}" in
+ '[' xzip = none ']'
+ compression_string='and compressing with '\''xzip:9'\'' with '\''4'\'' threads'
+ pixz -9 -p 4 /tmp/backups/redis_all_redis_20220414-120135.rdb
+ generate_checksum
+ var_true TRUE
+ '[' true = true ']'
+ '[' '' = 0 ']'
+ print_warn 'Skipping Checksum creation because backup did not complete successfully'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:45 [44m[WARN][49m /etc/services.available/10-db-backup/run ** [db-backup] Skipping Checksum creation because backup did not complete successfully
+ move_dbbackup
+ '[' '' = 0 ']'
+ print_warn 'Skipping moving DB Backup to final location because backup did not complete successfully'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:45 [44m[WARN][49m /etc/services.available/10-db-backup/run ** [db-backup] Skipping moving DB Backup to final location because backup did not complete successfully
+ rm -rf /tmp/backups/redis_all_redis_20220414-120135.rdb.xz
+ post_dbbackup all
++ date +%s
+ dbbackup_finish_time=1649962905
++ echo 10
+ dbbackup_total_time=10
+ var_true TRUE
+ '[' true = true ']'
+ print_notice 'Sending Backup Statistics to Zabbix'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:45 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Sending Backup Statistics to Zabbix
++ stat -c%s /backup/redis_all_redis_20220414-120135.rdb.xz
stat: can't stat '/backup/redis_all_redis_20220414-120135.rdb.xz': No such file or directory
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.size -o ''
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.size -o ''
Sending failed. Use option -vv for more detailed output.
++ date -r /backup/redis_all_redis_20220414-120135.rdb.xz +%s
date: can't stat '/backup/redis_all_redis_20220414-120135.rdb.xz': No such file or directory
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.datetime -o ''
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.datetime -o ''
Sending failed. Use option -vv for more detailed output.
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.status -o ''
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.status -o ''
Sending failed. Use option -vv for more detailed output.
++ echo 10
+ silent zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.backup_duration -o 10
+ '[' true = true ']'
+ zabbix_sender -c /etc/zabbix/zabbix_agentd.conf -k dbbackup.backup_duration -o 10
Sending failed. Use option -vv for more detailed output.
+ '[' -n '' ']'
+ '[' -d /assets/custom-scripts/ ']'
++ echo 10
++ awk '{printf "Hours: %d Minutes: %02d Seconds: %02d", $1/3600, ($1/60)%60, $1%60}'
+ print_notice 'DB Backup for '\''all'\'' time taken: Hours: 0 Minutes: 00 Seconds: 10'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:46 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] DB Backup for 'all' time taken: Hours: 0 Minutes: 00 Seconds: 10
++ date +%s
+ backup_finish_time=1649962906
++ echo 11
+ backup_total_time=11
+ '[' -z '' ']'
+ master_exit_code=0
++ date -d @1649962906 '+%Y-%m-%d %T %Z'
+ print_info 'Backup routines finish time: 2022-04-14 12:01:46 PDT with overall exit code 0'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:46 [42m[INFO][49m /etc/services.available/10-db-backup/run ** [db-backup] Backup routines finish time: 2022-04-14 12:01:46 PDT with overall exit code 0
++ echo 11
++ awk '{printf "Hours: %d Minutes: %02d Seconds: %02d", $1/3600, ($1/60)%60, $1%60}'
+ print_notice 'Backup routines time taken: Hours: 0 Minutes: 00 Seconds: 11'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:46 [100m[NOTICE][49m /etc/services.available/10-db-backup/run ** [db-backup] Backup routines time taken: Hours: 0 Minutes: 00 Seconds: 11
+ cleanup_old_data
+ '[' -n 8640 ']'
+ '[' 0 '!=' 1 ']'
+ print_info 'Cleaning up old backups'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:46 [42m[INFO][49m /etc/services.available/10-db-backup/run ** [db-backup] Cleaning up old backups
+ mkdir -p /backup
+ find /backup/ -mmin +8640 -iname '*' -exec rm '{}' ';'
+ var_true TRUE
+ '[' true = true ']'
+ print_debug 'Exitting due to manual mode'
+ output_off
+ '[' true = true ']'
+ set +x
2022-04-14.12:01:46 [45m[DEBUG][49m /etc/services.available/10-db-backup/run ** [db-backup] Exitting due to manual mode
+ exit 0
```
</details>

### Environment

- Image version / tag:  3.2.1
- Host OS: Ubuntu 21.10

Relevant docker-compose section:
```
  db-backup-redis:
    container_name: redis_backup
    image: tiredofit/db-backup:3.2.1
    volumes:
      - ./backups:/backup
    environment:
      - DEBUG_MODE=true
      - DB_TYPE=redis
      - DB_HOST=redis
      - DB_DUMP_FREQ=1440
      - DB_DUMP_BEGIN=0345
      - DB_CLEANUP_TIME=8640
      - COMPRESSION=XZ
      - COMPRESSION_LEVEL=9
      - TZ=America/Vancouver
    restart: unless-stopped
    depends_on:
      - redis
```

**Comment by tiredofit** [comment]
Thanks for the submission and confirmation. I am in a very limited internet area for the next few months - I will create a test environment here and see what I can do to resolve in short order.

**Comment by tiredofit** [comment]
I spent a couple hours here and released `3.3.0` which has considerable refinements to S3. At the same time after analyzing your logs I believe this is going to give you better Redis backups as well. 

Please, let me know if this resolves the backup issue.

**Comment by tiredofit** [comment]
Closing due to lack of response - assuming resolved.
